### PR TITLE
* fix execution with another active directory

### PR DESCRIPTION
**Pull request by rikkimongoose**
Fix for the following issue: https://github.com/bundler/bundler/issues/5030

MacOS Finder (and I sure, that some other shells well) run the script setting ~ as his active directory. Running **bin/console** or **bin/setup** from it, you'll receive the following error, because they are unable to find /lib/:

_/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require': cannot load such file --_

This fix helps them to find /lib/ and execute well.

### CODE REVIEW

File: lib/bundler/templates/newgem/bin/setup.tt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #!/usr/bin/env bash
 set -euo pipefail
+cd -
```
Comment (segiddins) [inline lib/bundler/templates/newgem/bin/setup.tt]: why is this change necessary?

Comment (rikkimongoose) [inline lib/bundler/templates/newgem/bin/setup.tt]: Because shells like MacOS Finder produce the same problem to setup as well. They execute it with ~ as active directory and bundle install command just can't find proper files.

With this line I say to script "please, set the directory you're in as the current one". I've found "cd -" one-liner a long time ago somewhere in bash manuals.

Comment (rikkimongoose) [inline lib/bundler/templates/newgem/bin/setup.tt]: Checked it at my Ubuntu and found out that cd - is Mac-only shotcut.

Updated it with much more portable and accurate version, borrowed from here:

http://stackoverflow.com/questions/3349105/how-to-set-current-working-directory-to-the-directory-of-the-script

Comment (segiddins) [inline lib/bundler/templates/newgem/bin/setup.tt]: `cd -` just changes to the last directory

Comment (rikkimongoose) [inline lib/bundler/templates/newgem/bin/setup.tt]: Yes, it was my mistake.

Please, check the following commit, I've wrote a proper version of cd.

Comment (segiddins) [inline lib/bundler/templates/newgem/bin/setup.tt]: I'm sorry but I still don't understand why changing directories at all is necessary here

Comment (rikkimongoose) [inline lib/bundler/templates/newgem/bin/setup.tt]: Just try this on a Unix or Mac system:

```bash
mkdir ~/dev
cd ~/dev
bundle gem foo
# edit gemspec here, and after....
cd ~
# let's try to install
~/dev/foo/bin/setup
```

It will say "Could not locate Gemfile or .bundle/ directory", because active directory will be ~.

**Comment by segiddins** [comment]
@indirect thoughts?

**Comment by indirect** [comment]
This bug is legit, but I'm not sure yet if this patch is the best solution. I think the solution for `bin/console` is to instead add this line before `require 'bundler/setup'`:

```ruby
BUNDLE_GEMFILE = File.join("Gemfile", File.expand_path("..", __FILE__))
```

Bundler knows to add the `lib` from this gem to the loadpath, but it needs to be able to find the Gemfile for that to happen.

As for `bin/setup`, I'm not sure yet. It seems like getting the directory of the running script from bash is [super scary and unreliable](http://www.ostricher.com/2014/10/the-right-way-to-get-the-directory-of-a-bash-script/).

That said, I definitely think we should make sure that both `bin/console` and `bin/setup` work when executed from any working directory.

**Comment by colby-swandale** [comment]
ping @rikkimongoose

**Comment by esasse** [comment]
#5030 was closed and this PR is not active in a long time as well.
